### PR TITLE
Tooltip: remove unecessary delay in tests except where needed

### PR DESCRIPTION
**Pull request by DaniGuardiola**
## What?

There was a need to add a delay after every Tooltip test. After an Ariakit update, it's not necessary anymore for every test. See discussion: https://github.com/WordPress/gutenberg/pull/57345#discussion_r1435167187

## Why?

Faster tests.

## How?

Removed delay after every test and only added where necessary (after one specific test).

## Testing Instructions

Run Tooltip unit tests.

### Testing Instructions for Keyboard

N/A

## Screenshots or screencast 

N/A

### CODE REVIEW

File: packages/components/CHANGELOG.md [hunk 1]
```diff
@@ -8,12 +8,13 @@
 
 ### Enhancements
 
+-   `Tooltip`: Make tests faster ([#60897](https://github.com/WordPress/gutenberg/pull/60897)).
 -   `ExternalLink`: Use unicode arrow instead of svg icon ([#60255](https://github.com/WordPress/gutenberg/pull/60255)).
 -   `ProgressBar`: Move the indicator width styles from emotion to a CSS variable ([#60388](https://github.com/WordPress/gutenberg/pull/60388)).
--   `Text`: Add `text-wrap: pretty;` to improve wrapping. ([#60164](https://github.com/WordPress/gutenberg/pull/60164)).
--   `Navigator`: Navigation to the active path doesn't create a new location history. ([#60561](https://github.com/WordPress/gutenberg/pull/60561))
--   `FormToggle`: Forwards ref to input. ([#60234](https://github.com/WordPress/gutenberg/pull/60234)).
--   `ToggleControl`: Forwards ref to FormToggle. ([#60234](https://github.com/WordPress/gutenberg/pull/60234)).
```
Comment (tyxla) [inline packages/components/CHANGELOG.md:16]: Thanks for cleaning those up 🙌

File: packages/components/src/tooltip/test/index.tsx [hunk 2]
```diff
@@ -312,6 +304,11 @@ describe( 'Tooltip', () => {
 			// Hover outside of the anchor, tooltip should hide
 			await hoverOutside();
 			await waitExpectTooltipToHide();
+
+			// Prevent this test from interfering with the next one.
+			// "Tooltips appear instantly if another tooltip has just been hidden."
+			// See: https://github.com/WordPress/gutenberg/pull/57345#discussion_r1435495655
+			await sleep( 3000 );
```
Comment (tyxla) [inline packages/components/src/tooltip/test/index.tsx:311]: Ideally, there would be a better way, and something more specific to wait for, but this is not related to the PR, so 🤷

**Review by tyxla — APPROVED** [review]
Nice, thanks 👍 

Let's keep an eye out for flakiness 👁️

File: packages/components/src/tooltip/test/index.tsx (hunk 2 above)
Comment (diegohaz) [reply packages/components/src/tooltip/test/index.tsx:311]: I believe this won't be needed with `@ariakit/react@^0.4.2`.

Comment (DaniGuardiola) [reply packages/components/src/tooltip/test/index.tsx:311]: Oh I didn't realize we're still on 0.3.12, is there any discussion on upgrading? Any relevant blockers? @tyxla

Comment (tyxla) [reply packages/components/src/tooltip/test/index.tsx:311]: I think there were some `v0.4.0` breaking changes we need to address, like:

- We may have to manually set `role="gridcell"` to grid cells within `Composite` when  using `role="grid"`
- Data attributes can no longer be empty strings
- Type changes

Nothing too critical though - I'd go through all the breaking changes and if there are any that apply to the package, verify that we update properly without breaking something. 

Also I haven't been directly involved with updating Ariakit before, so perhaps @mirka might know more.

Comment (tyxla) [reply packages/components/src/tooltip/test/index.tsx:311]: For the protocol, I'd love for us to update to the latest version. There are so many improvements and new features we could benefit from!
